daemon/create_unix.go

```diff
@@ -26,8 +27,12 @@ func (daemon *Daemon) createContainerOSSpecificSettings(container *container.Con
 	}
 	defer daemon.Unmount(container)
 
-	rootIDs := daemon.idMapping.RootPair()
-	if err := container.SetupWorkingDirectory(rootIDs); err != nil {
+	user, err := getUser(container, config.User)
+	if err != nil {
+		return err
+	}
+	owner := idtools.Identity{UID: int(user.UID), GID: int(user.GID)}
```

Comment:
I think the `SetupWorkingDirectory` is ran within the "host's" context; in that case, I'm wondering if this needs to remap the UID/GID to match the container's user in case user-namespaces are enabled.

`daemon.idMapping.RootPair()` calls `GetRootUIDGID()`, which calls `toHost()` to perform the mapping from "container user" to "host user"; https://github.com/moby/moby/blob/0a3336fd7d19f7114fce2ff849a8989ed33e2059/pkg/idtools/idtools.go#L60-L65

There is an `IdentityMapping.ToHost()` method (but it expects `UIDMaps` to be propagated, so we need to make sure that's the case); https://github.com/moby/moby/blob/0a3336fd7d19f7114fce2ff849a8989ed33e2059/pkg/idtools/idtools.go#L133

Reply:
tested with daemon configured with userns-remap, and indeed need to convert Identity to host